src/_pytest/assertion/util.py

```diff
@@ -308,6 +311,21 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> list[str]:
                 ]
                 left = left[:-i]
                 right = right[:-i]
+        shortest = min(left, right, key=lambda x: len(x))
+        lines = j = start = 0
+        if shortest.count("\n") >= DEFAULT_MAX_LINES:
```

Comment:
I'm under the impression the code can be made simpler by skipping max chars first, then dealing with max lines... :thinking: